src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -601,22 +587,18 @@ private void MakeExplicitParameterTypeInferences(Binder binder, BoundExpression
             }
             else if (argument.Kind == BoundKind.TupleLiteral)
             {
-                MakeExplicitParameterTypeInferences(binder, (BoundTupleLiteral)argument, target, isExactInference, ref useSiteDiagnostics);
+                MakeExplicitParameterTypeInferences(binder, (BoundTupleLiteral)argument, isNullable, target, kind, ref useSiteDiagnostics);
             }
-            else if (IsReallyAType(source))
+            else
             {
-                if (isExactInference)
-                {
-                    ExactInference(source, target, ref useSiteDiagnostics);
-                }
-                else
+                if (IsReallyAType(source))
                 {
-                    LowerBoundInference(source, target, ref useSiteDiagnostics);
+                    ExactOrBoundsInference(kind, TypeSymbolWithAnnotations.Create(source, isNullable), target, ref useSiteDiagnostics);
                 }
             }
         }
 
-        private void MakeExplicitParameterTypeInferences(Binder binder, BoundTupleLiteral argument, TypeSymbol target, bool isExactInference, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        private void MakeExplicitParameterTypeInferences(Binder binder, BoundTupleLiteral argument, bool? isNullable, TypeSymbolWithAnnotations target, ExactOrBoundsKind kind, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
```

Comment:
Consider breaking long lines (also above).

Reply:
Normally I would break long lines but there was enough change in this file already.